internal/rbd/rbd_util.go

```diff
@@ -1103,12 +1125,18 @@ func (rv *rbdVolume) cloneRbdImageFromSnapshot(ctx context.Context, pSnapOpts *r
 		return fmt.Errorf("failed to set image features: %w", err)
 	}
 
+	// As the clone is yet to be created, open the Ioctx.
 	err = rv.openIoctx()
 	if err != nil {
 		return fmt.Errorf("failed to get IOContext: %w", err)
 	}
+	// Destroy only the ioctx, connection is needed for other operations.
+	defer func() {
+		defer rv.ioctx.Destroy()
```

Comment:
No need to do this, keep it open for potential other needs. When `rv.Destroy()` gets called, all attributes of `rv` get cleaned up.